### PR TITLE
Navigation us3

### PR DESCRIPTION
**Pull request by LHJE**
closes #75 

***Contributors***
Luke

***What This Does***
Ensures logged in users can see a link to their profile page, to the logout link, and that they do not see the log in or register link when logged in.
It also ensures it shows the logged in user a notifying message signifying who they are logged in as.

Note:
This holds the commit information regarding the commits for the US's we've already worked on.  This branch's new commits start at commit 72fdbea.

### CODE REVIEW

**Comment by leahriffell** [comment]
Thanks for working on this Luke! I just have a question on this git flow and how we could avoid seeing changes that are actually tied to a different milestone reflecting as new changes here. 

Priya did you run into this in last time you used this git flow? If we wait till a milestone's changes are merged to dev, will this avoid this? Looks like we could maybe have a lot of merge conflicts once we do merge `navigation` to `development`.

File: spec/features/navigation_spec.rb [hunk 1]
```diff
@@ -3,6 +3,20 @@
 
 RSpec.describe 'Site Navigation' do
   describe 'As a Visitor' do
+    before :each do
+      @user = User.create(name:"Jackie Chan", address:"skdjfhdskjfh", city:"kajshd", state:"jsdh", zip:"88888", email: "tombroke@gmail.com", password:"Iamapassword", password_confirmation:"Iamapassword", role: 0)
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:7]: Perfect user. Love it

File: spec/features/navigation_spec.rb [hunk 2]
```diff
@@ -3,6 +3,20 @@
...
+      within 'nav' do
+        click_link "Login"
+      end
+
+      fill_in :email, with: @user.email
+      fill_in  :password, with: @user.password
+      click_button "Login"
+    end
+
     it "I see a nav bar with links to all pages" do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:20]: These were old tests?

File: spec/features/navigation_spec.rb (hunk 1 above)
Comment (priyapower) [reply spec/features/navigation_spec.rb:7]: I also like that we are utilizing the before :each as often as we can

File: spec/features/navigation_spec.rb [hunk 3]
```diff
@@ -33,5 +47,38 @@
       end
 
     end
+
+    it "I can see a profile link if logged in on all pages" do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:51]: This test and below meet us3 requirement that a user sees "login" and "profile"

File: spec/features/navigation_spec.rb [hunk 4]
```diff
@@ -33,5 +47,38 @@
...
+
+    it "I can see a logout link if logged in on all pages" do
+      visit '/merchants'
+
+      within 'nav' do
+        expect(page).to have_link("Logout")
+      end
+    end
+
+    it "I cannot see a login or register link if logged in on all pages" do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:67]: This test meets us3 requirements of NOT seeing "login" or "register" anymore

File: spec/features/navigation_spec.rb (hunk 1 above)
Comment (priyapower) [inline spec/features/navigation_spec.rb:7]: Perfect user. Love it

File: spec/features/navigation_spec.rb [hunk 5]
```diff
@@ -3,6 +3,20 @@
 
 RSpec.describe 'Site Navigation' do
   describe 'As a Visitor' do
+    before :each do
+      @user = User.create(name:"Jackie Chan", address:"skdjfhdskjfh", city:"kajshd", state:"jsdh", zip:"88888", email: "tombroke@gmail.com", password:"Iamapassword", password_confirmation:"Iamapassword", role: 0)
+
+      visit '/merchants'
+
+      within 'nav' do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:11]: I also like that we are utilizing the before :each as often as we can

File: spec/features/navigation_spec.rb (hunk 2 above)
Comment (priyapower) [inline spec/features/navigation_spec.rb:20]: These were old tests?

File: spec/features/navigation_spec.rb (hunk 3 above)
Comment (priyapower) [inline spec/features/navigation_spec.rb:51]: This test meets us3 requirement that a user sees "profile"

File: spec/features/navigation_spec.rb [hunk 6]
```diff
@@ -33,5 +47,38 @@
...
+
+    it "I can see a profile link if logged in on all pages" do
+      visit '/merchants'
+
+      within 'nav' do
+        expect(page).to have_link("Profile")
+      end
+    end
+
+    it "I can see a logout link if logged in on all pages" do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:59]: This test meets us3 requirement that a user sees "logout"

File: spec/features/navigation_spec.rb (hunk 4 above)
Comment (priyapower) [inline spec/features/navigation_spec.rb:67]: This test meets us3 requirements of NOT seeing "login" or "register" anymore

File: spec/features/navigation_spec.rb [hunk 7]
```diff
@@ -33,5 +47,38 @@
...
+    it "I cannot see a login or register link if logged in on all pages" do
+      visit '/merchants'
+
+      within 'nav' do
+        expect(page).to_not have_link("Login")
+        expect(page).to_not have_link("Register")
+      end
+    end
+
+    it "I can see 'logged in as' message if logged in on all pages" do
```
Comment (priyapower) [inline spec/features/navigation_spec.rb:76]: This test meets us3 requirements for flash message "logged in as <dynamic name>"

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -10,10 +10,20 @@
 
   <body>
     <nav class = "topnav">
+
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Register", "/register"%>
+      <%= link_to "Profile", "/profile"%>
+      <% if !current_user %>
```
Comment (priyapower) [inline app/views/layouts/application.html.erb:17]: Loving this switch statement to ensure our logged in users only see the links that make sense

**Review by priyapower — APPROVED** [review]
Love it! I think it is ready to merge

File: spec/features/navigation_spec.rb (hunk 1 above)
Comment (LHJE) [reply spec/features/navigation_spec.rb:7]: The besssst.

File: spec/features/navigation_spec.rb (hunk 2 above)
Comment (LHJE) [reply spec/features/navigation_spec.rb:20]: Yes!  These ones came with the original build.

**Pull request merged**